python/erdos/operator.py

```diff
@@ -14,8 +14,10 @@
 
 MAX_NUM_RUNTIME_SAMPLES = 1000
 
+T = TypeVar("T")
 
-class BaseOperator:
+
+class BaseOperator(Generic[T]):
```

Comment:
```suggestion
class BaseOperator:
```

The `BaseOperator` shouldn't be typed.